### PR TITLE
Removing HTTP requests from Bucket logging methods.

### PR DESCRIPTION
**Pull request by dhermes**
Relates to  #728

### CODE REVIEW

File: gcloud/storage/test_bucket.py [hunk 1]
```diff
@@ -758,6 +758,7 @@ def test_get_logging_w_prefix(self):
         resp_to_reload = before
         connection = _Connection(resp_to_reload)
         bucket = self._makeOne(NAME, connection)
+        bucket._reload_properties()
```
Comment (tseaver) [inline gcloud/storage/test_bucket.py]: Given that we aren't relying on implicit GET / PATCH any longer, shouldn't the tests just drop the whole connection dance, and setup / make assertions only about local state?

Comment (dhermes) [inline gcloud/storage/test_bucket.py]: Yes. I was going for the fly-by / low-hanging fruit approach of test-fixes.

I can cover it here if you like. Or, instead, I can file a bug to do a standalone test cleanup (essentially just `git grep '.patch'` and `git grep '._reload'`)

**Comment by dhermes** [comment]
@tseaver based on https://github.com/GoogleCloudPlatform/gcloud-python/pull/773#issuecomment-87799124 are we a go to merge this?

**Comment by dhermes** [comment]
@tseaver Just rebased (the public `.reload()` in #770 made this un-mergeable).

Are we good?

**Comment by tseaver** [comment]
LGTM:  I added #778 to track the testcase cleanups.

**Pull request merged**